src/ui/components/search/searchcomponent.js

```diff
@@ -348,26 +378,58 @@ export default class SearchComponent extends Component {
     if (this.iconState === iconState) {
       return;
     }
+    this.prevState = this.iconState;
     this.iconState = iconState;
     if (!this.isRequestingAnimationFrame) {
       this.isRequestingAnimationFrame = true;
       window.requestAnimationFrame(() => {
-        this.forwardIcon.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
-        this.reverseIcon.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
+        const iconEls = DOM.queryAll(this._container, '.js-yxt-SearchBar-Icon');
+        iconEls.forEach((el) => {
+          el.classList.contains('js-yxt-SearchBar-LoadingIndicator')
```

Comment:
I know I didn't do this for the original animated search icon, but would we be able to move the state management into another class? If this was React we would probably have it be another component, but child components do not work nicely in the SDK so we're not able to divide responsibility like that. It's kind of hard to tell what's going on here. I think generally speaking it's cleaner to keep state as some javascript variables and not rely on css classes in the DOM, otherwise you can end up in jQuery-like spaghetti land